src/CppUTest/SimpleString.cpp

```diff
@@ -478,6 +478,40 @@ SimpleString HexStringFrom(unsigned long value)
     return StringFromFormat("%lx", value);
 }
 
+#ifdef CPPUTEST_USE_LONG_LONG
+
+SimpleString StringFrom(cpputest_longlong value)
+{
+    return StringFromFormat("%lld", value);
+}
+
+SimpleString StringFrom(cpputest_ulonglong value)
+{
+    return StringFromFormat("%llu (0x%llx)", value, value);
+}
+
+SimpleString HexStringFrom(cpputest_longlong value)
+{
+    return StringFromFormat("%llx", value);
+}
+
+SimpleString HexStringFrom(cpputest_ulonglong value)
+{
+    return StringFromFormat("%llx", value);
+}
+
+SimpleString HexStringFrom(const void* value)
+{
+    return HexStringFrom((cpputest_ulonglong) value);
+}
+
+SimpleString HexStringFrom(void (*value)())
+{
+    return HexStringFrom((cpputest_ulonglong) value);
+}
+
+#else   /* CPPUTEST_USE_LONG_LONG */
+
 static long convertPointerToLongValue(const void* value)
```

Comment:
They're defined differently in the `#else` path (as originally defined prior to longlong support).

Basically, if `long long` is enabled then it converts pointers to longlongs prior to formatting to avoid possibly trying to convert a 64-bit pointer to a 32-bit long (on Windows), as previously discussed as a problem in `convertPointerToLongValue` just below.

On Linux this might result in unnecessarily converting a 64-bit pointer to a 128-bit integer prior to formatting, but other than marginal performance loss this should be harmless.

It does assume that `unsigned long long` is the same size or larger than a pointer on all platforms, but as far as I am aware that is true on all current platforms.  In the event it fails to be true on some future platform (which seems unlikely) then it should generate a warning or error in that case and we can worry about it then.